### PR TITLE
Allowing more iconsizes

### PR DESCRIPTION
**Pull request by aminya**
- Adds icons size of 18,21,28
- Setting default size to 21

Derived from https://github.com/suda/tool-bar/pull/218

### CODE REVIEW

**Comment by aminya** [comment]
Could you review this? @suda

File: README.md [hunk 1]
```diff
@@ -33,8 +33,11 @@ On which edge of the editor should the tool bar appear. Possible options:
 ### Icon size
 
 *   Very Small *(12px)*
-*   Small *(16px)*
+*   Smaller *(16px)*
+*   Small: *(18px)*
```
Comment (ericcornelissen) [inline README.md]: Perhaps change this to :arrow_down:?

```markdown
*   Tiny *(12px)*
*   Vary small *(16px)*
*   Small: *(18px)*
```

As far as I know changing these names doesn't break backwards compatibility.

File: lib/tool-bar.js [hunk 2]
```diff
@@ -29,8 +29,8 @@ exports.config = {
   },
   iconSize: {
     type: 'string',
-    default: '24px',
-    enum: ['12px', '16px', '24px', '32px'],
+    default: '21px',
```
Comment (ericcornelissen) [inline lib/tool-bar.js]: I'm personally not sure if changing the default is a good idea (also given my comment above).

**Comment by aminya** [comment]
> 
> 
> Hi @aminya, I'm not a maintainer but I'm just trying to help @suda out with reviews.
> 
> The implementation looks good in theory, but for some reason I'm getting these weird "blocks" around my buttons **only** for the new sizes you added. See an example below: left is 21px, right is 24px 🤔 Do you have them too? Any ideas what might be causing them?
> 
> preview
> 
> I'm using:
> 
>     * UI theme: [atom-material-ui](https://atom.io/themes/atom-material-ui)
> 
>     * Syntax: [one-dark-material-syntax](https://atom.io/themes/one-dark-material-syntax)
> 
>     * Toolbar: [flex-tool-bar](https://atom.io/packages/flex-tool-bar)

I don't have these. Maybe this problem. I tried with different themes, and it was fine.

File: README.md (hunk 1 above)
Comment (aminya) [reply README.md]: Fixed in https://github.com/suda/tool-bar/pull/277/commits/3489f07427c703faaf7f2425a6a41c94dc82923d

File: lib/tool-bar.js (hunk 2 above)
Comment (aminya) [reply lib/tool-bar.js]: I don't feel strongly about it. We can return to 24. Whatever you guys think.

**Comment by ericcornelissen** [comment]
Thanks for the response @aminya, I investigated further and it turns out the discrepancy is caused by [atom-material-ui](https://atom.io/themes/atom-material-ui) (it removes margins for existing button sizes but not for the new button sizes) so you don't need to worry about it 👍 


> Your photo doesn't look bad though...

Not too bad, but definitely worse in my opinion, but that is objective. Subjectively speaking, it is very weird that the `21px` buttons take up **more** space than the `24px` buttons.

**Comment by aminya** [comment]
> 
> 
> Thanks for the response @aminya, I investigated further and it turns out the discrepancy is caused by [atom-material-ui](https://atom.io/themes/atom-material-ui) (it removes margins for existing button sizes but not for the new button sizes) so you don't need to worry about it 👍
> 
> > Your photo doesn't look bad though...
> 
> Not too bad, but definitely worse in my opinion, but that is objective. Subjectively speaking, it is very weird that the `21px` buttons take up **more** space than the `24px` buttons.


Atom material looks very bulky for me. It has more problems that make it the worse theme for me. 
- It doesn't show the tab names correctly (just squashed together)
- The project management is large and bulky for no reason
- The settings seem just cluttered
- New one: It doesn't show 21px toolbar buttons correctly.

I was never able to use it, and now it is even worse. 

I love One-Dark.

Compare One-Dark:
image

To Material:
image

**Comment by aminya** [comment]
They don't pay attention to the issues: https://github.com/atom-material/atom-material-ui/issues/483, so I guess we should just ignore the problem with Meterial

**Comment by ericcornelissen** [comment]
> They don't pay attention to the issues: [atom-material/atom-material-ui#483](https://github.com/atom-material/atom-material-ui/issues/483), so I guess we should just ignore the problem with Material

I don't think we should it into account in the context of this Pull Request.


> I can add a warning about the issues with Material UI to the readme, or even a simple code that detects if the user is using Atom-UI-Meterial and giving warning about that.

Not sure if that is needed, I personally don't think so. If you think it is very easy to do you can share some code that users can put in their personal scripts/stylesheets to fix the problem locally?

**Comment by aminya** [comment]
> 
> Not sure if that is needed, I personally don't think so. If you think it is very easy to do you can share some code that users can put in their personal scripts/stylesheets to fix the problem locally?

Yes the warning code is simple. I added the it.

File: lib/tool-bar.js (hunk 2 above)
Comment (suda) [reply lib/tool-bar.js]: I'd definitely keep the 24px default. There's no reason for changing it.

File: lib/tool-bar-view.js [hunk 3]
```diff
@@ -4,6 +4,11 @@ const rafDebounce = require('./raf-debounce');
 module.exports = class ToolBarView {
 
   constructor () {
+    // warning for Atom-Material-UI
+    if (atom.themes.getEnabledThemeNames()[1] === 'atom-material-ui') {
```
Comment (suda) [inline lib/tool-bar-view.js]: I'm not fond of this check. Can we have a CSS override for `atom-material-ui` in this package if they are not willing to fix it?

**Comment by aminya** [comment]
I addressed the review comments. This is now ready to get merged.

**Pull request merged**